tests/test_rio_create.py

```diff
@@ -0,0 +1,354 @@
+            "3",
+            "-h",
+            "512",
+            "-w",
+            "256",
+            "--nodata",
+            "255",
+            outfile,
+        ],
+    )
+    assert result.exit_code == 0
+
+    with rasterio.open(outfile) as dataset:
+        assert dataset.shape == (512, 256)
+        assert dataset.count == 3
+        assert dataset.dtypes == ("uint8", "uint8", "uint8")
+        assert dataset.nodatavals == (255, 255, 255)
+        assert dataset.driver == "GTiff"
+        raster = dataset.read(masked=True)
+        assert (raster.data == 255).all()
+        assert raster.mask.all()
+
+
+def test_create_creation_opts(tmp_path, runner):
+    """Create a new tif with creation/opening options."""
+    outfile = str(tmp_path.joinpath("out.tif"))
+    result = runner.invoke(
+        main_group,
+        [
+            "create",
+            "-f",
+            "GTiff",
+            "-t",
+            "uint8",
+            "-n",
+            "3",
+            "-h",
+            "512",
+            "-w",
+            "256",
+            "--co",
+            "tiled=true",
+            "--co",
+            "blockxsize=128",
+            "--co",
+            "blockysize=256",
+            outfile,
+        ],
+    )
+    assert result.exit_code == 0
+
+    with rasterio.open(outfile) as dataset:
+        assert dataset.shape == (512, 256)
+        assert dataset.count == 3
+        assert dataset.dtypes == ("uint8", "uint8", "uint8")
+        assert dataset.driver == "GTiff"
+        assert all(((256, 128) == hw for hw in dataset.block_shapes))
+
+
+def test_create_no_overwrite(tmp_path, runner):
+    """Don't allow overwrite of existing file without option."""
+    outpath = tmp_path.joinpath("out.tif")
+    outpath.touch()
+    outfile = str(outpath)
+
+    result = runner.invoke(
+        main_group,
+        [
+            "create",
+            "-f",
+            "GTiff",
+            "-t",
+            "uint8",
+            "-n",
+            "3",
+            "-h",
+            "512",
+            "-w",
+            "256",
+            outfile,
+        ],
+    )
+    assert result.exit_code == 1
+    assert "File exists and won't be overwritten" in result.output
+
+
+def test_create_overwrite(tmp_path, runner):
+    """Allow overwrite of existing file with option."""
+    outpath = tmp_path.joinpath("out.tif")
+    outpath.touch()
+    outfile = str(outpath)
+
+    result = runner.invoke(
+        main_group,
+        [
+            "create",
+            "-f",
+            "GTiff",
+            "-t",
+            "uint8",
+            "-n",
+            "3",
+            "-h",
+            "512",
+            "-w",
+            "256",
+            "--overwrite",
+            outfile,
+        ],
+    )
+    assert result.exit_code == 0
+
+
+def test_create_no_overwrite_nonfile(runner):
+    """Don't allow overwrite of existing non-file without option."""
+    with MemoryFile(bytes(bytearray(100000))) as memfile:
```

Comment:
This is a hack to create a "file" that can't be found by `os.path` but can be found by GDAL.